### ISSUE TITLE
TypeError: set_style() got an unexpected keyword argument 'stroke_opacity'

### ISSUE THREAD

**Issue by emonhossainraihan** [ManimCommunity/manim#1388]
## Description of bug / unexpected behavior

I have been working with VectorScene over the past few days and have encountered a number of bugs. Some method of VectorScene didn't work properly, e.g., `label_vector`, `vector_to_coords` and `show_ghost_movement`. It gives same the error 

> TypeError: set_style() got an unexpected keyword argument 'stroke_opacity'

When I tried to trace the error I get this in the Traceback:

```
│ File "/home/emonhossain/.local/lib/python3.8/site-packages/manim/mobject/type│
│s/vectorized_mobject.py", line 281, in match_style                            │
│     278         return ret                                                   │
│     279                                                                      │
│     280     def match_style(self, vmobject, family=True):                    │
│  ❱  281         self.set_style(**vmobject.get_style(), family=False) 
```
I didn't find where `stroke_opacity` was passed. Because I wasn't pass any configuration `key-value` pair in any of the method.

## Expected behavior

I was expecting two labeled vector mobject will be added on the Scene for `VectorExpment`.

## How to reproduce the issue

<details><summary>Code for reproducing the problem</summary>

```py
class VectorExpment(VectorScene):
    def construct(self):
        self.add_plane()
        v1 = self.add_vector([1, 1])
        v2 = self.add_vector([2, 2])
        l1 = self.label_vector(v1,"a")
        l2 = self.label_vector(v2,"b")
        self.add(v1,v2,l1,l2)
        self.wait()

class VectorCoordExpment(VectorScene):
    def construct(self):
        v1 = self.add_vector([1, 1])
        coords1, x_line1, y_line1 = self.vector_to_coords(v1)
        self.wait()
```

</details>

## Logs
<details><summary>Terminal output</summary>

```
Manim Community v0.5.0
[04/22/21 13:32:27] DEBUG    Hashing ...                                                   hashing.py:239
                    DEBUG    Hashing done in 0.063073 s.                                   hashing.py:252
                    DEBUG    Hash generated :  450974505_413878538_1896313505              hashing.py:256
                    DEBUG    List of the first few animation hashes of the scene:   cairo_renderer.py:107
                             ['450974505_413878538_1896313505']                                          
[04/22/21 13:32:29] INFO     Animation 0 : Partial movie file written in {'/home scene_file_writer.py:395
                             /emonhossain/Desktop/videos/media/videos/experiment_VScene                         
                             /1080p60/partial_movie_files/VectorExpment/45097                         
                             4505_413878538_1896313505.mp4'}                                             
                    DEBUG    Hashing ...                                                   hashing.py:239
                    DEBUG    Hashing done in 0.063201 s.                                   hashing.py:252
                    DEBUG    Hash generated :  2788726626_1012985844_1722812649            hashing.py:256
                    DEBUG    List of the first few animation hashes of the scene:   cairo_renderer.py:107
                             ['450974505_413878538_1896313505',                                          
                             '2788726626_1012985844_1722812649']                                         
[04/22/21 13:32:31] INFO     Animation 1 : Partial movie file written in {'/home scene_file_writer.py:395
                             /emonhossain/Desktop/videos/media/videos/experiment_VScene                         
                             /1080p60/partial_movie_files/VectorExpment/27887                         
                             26626_1012985844_1722812649.mp4'}                                           
                    INFO     Writing "\vec{\textbf{a}}" to                         tex_file_writing.py:81
                             media/Tex/ab8db85638720937.tex                                              
[04/22/21 13:32:32] DEBUG    Hashing ...                                                   hashing.py:239
                    DEBUG    Hashing done in 0.073055 s.                                   hashing.py:252
                    DEBUG    Hash generated :  2788726626_3584723761_2346814504            hashing.py:256
                    DEBUG    List of the first few animation hashes of the scene:   cairo_renderer.py:107
                             ['450974505_413878538_1896313505',                                          
                             '2788726626_1012985844_1722812649',                                         
                             '2788726626_3584723761_2346814504']                                         
Traceback (most recent call last):
╭──────────────────────────────────────────────────────────────────────────────────────╮
│ File "/home/emonhossain/.local/lib/python3.8/site-packages/manim/__main__.py", line  │
│76, in main                                                                           │
│    73             for SceneClass in scene_classes_from_file(input_file):             │
│    74                 try:                                                           │
│    75                     scene = SceneClass()                                       │
│  ❱ 76                     scene.render()                                             │
│    77                 except Exception:                                              │
│    78                     console.print_exception()                                  │
│    79                                                                                │
│ File "/home/emonhossain/.local/lib/python3.8/site-packages/manim/scene/scene.py",    │
│line 183, in render                                                                   │
│     180         """                                                                  │
│     181         self.setup()                                                         │
│     182         try:                                                                 │
│  ❱  183             self.construct()                                                 │
│     184         except EndSceneEarlyException:                                       │
│     185             pass                                                             │
│     186         self.tear_down()                                                     │
│ File "/home/emonhossain/Desktop/videos/experiment_VScene.py", line 1190, in construct    │
│    1187         self.add_plane()                                                     │
│    1188         v1 = self.add_vector([1, 1])                                         │
│    1189         v2 = self.add_vector([2, 2])                                         │
│  ❱ 1190         l1 = self.label_vector(v1,"a")                                       │
│    1191         l2 = self.label_vector(v2,"b")                                       │
│    1192         self.add(v1,v2,l1,l2)                                                │
│    1193         self.wait()                                                          │
│ File "/home/emonhossain/.local/lib/python3.8/site-packages/manim/scene/vector_space_s│
│cene.py", line 347, in label_vector                                                   │
│     344         """                                                                  │
│     345         label = self.get_vector_label(vector, label, **kwargs)               │
│     346         if animate:                                                          │
│  ❱  347             self.play(Write(label, run_time=1))                              │
│     348         self.add(label)                                                      │
│     349         return label                                                         │
│     350                                                                              │
│ File "/home/emonhossain/.local/lib/python3.8/site-packages/manim/scene/scene.py",    │
│line 803, in play                                                                     │
│     800             return np.max([animation.run_time for animation in animations])  │
│     801                                                                              │
│     802     def play(self, *args, **kwargs):                                         │
│  ❱  803         self.renderer.play(self, *args, **kwargs)                            │
│     804                                                                              │
│     805     def wait(self, duration=DEFAULT_WAIT_TIME, stop_condition=None):         │
│     806         self.play(Wait(duration=duration, stop_condition=stop_condition))    │
│ File "/home/emonhossain/.local/lib/python3.8/site-packages/manim/renderer/cairo_rende│
│rer.py", line 116, in play                                                            │
│    113         self.static_image = self.save_static_frame_data(scene, scene.static_mo│
│    114                                                                               │
│    115         self.file_writer.begin_animation(not self.skip_animations)            │
│  ❱ 116         scene.begin_animations()                                              │
│    117         if scene.is_current_animation_frozen_frame():                         │
│    118             self.update_frame(scene)                                          │
│    119             # self.duration stands for the total run time of all the animation│
│ File "/home/emonhossain/.local/lib/python3.8/site-packages/manim/scene/scene.py",    │
│line 875, in begin_animations                                                         │
│     872     def begin_animations(self) -> None:                                      │
│     873         """Start the animations of the scene."""                             │
│     874         for animation in self.animations:                                    │
│  ❱  875             animation.begin()                                                │
│     876                                                                              │
│     877     def is_current_animation_frozen_frame(self) -> bool:                     │
│     878         """Returns wether the current animation produces a static frame (gene│
│ File                                                                                 │
│"/home/emonhossain/.local/lib/python3.8/site-packages/manim/animation/creation.py",   │
│line 244, in begin                                                                    │
│    241                                                                               │
│    242     def begin(self) -> None:                                                  │
│    243         self.outline = self.get_outline()                                     │
│  ❱ 244         super().begin()                                                       │
│    245                                                                               │
│    246     def get_outline(self) -> Mobject:                                         │
│    247         outline = self.mobject.copy()                                         │
│ File                                                                                 │
│"/home/emonhossain/.local/lib/python3.8/site-packages/manim/animation/animation.py",  │
│line 94, in begin                                                                     │
│     91             # the internal updaters of self.starting_mobject,                 │
│     92             # or any others among self.get_all_mobjects()                     │
│     93             self.mobject.suspend_updating()                                   │
│  ❱  94         self.interpolate(0)                                                   │
│     95                                                                               │
│     96     def finish(self) -> None:                                                 │
│     97         self.interpolate(1)                                                   │
│ File                                                                                 │
│"/home/emonhossain/.local/lib/python3.8/site-packages/manim/animation/animation.py",  │
│line 143, in interpolate                                                              │
│    140     # Methods for interpolation, the mean of an Animation                     │
│    141     def interpolate(self, alpha: float) -> None:                              │
│    142         alpha = np.clip(alpha, 0, 1)                                          │
│  ❱ 143         self.interpolate_mobject(self.rate_func(alpha))                       │
│    144                                                                               │
│    145     def update(self, alpha: float) -> None:                                   │
│    146         """                                                                   │
│ File                                                                                 │
│"/home/emonhossain/.local/lib/python3.8/site-packages/manim/animation/animation.py",  │
│line 160, in interpolate_mobject                                                      │
│    157         families = list(self.get_all_families_zipped())                       │
│    158         for i, mobs in enumerate(families):                                   │
│    159             sub_alpha = self.get_sub_alpha(alpha, i, len(families))           │
│  ❱ 160             self.interpolate_submobject(*mobs, sub_alpha)                     │
│    161                                                                               │
│    162     def interpolate_submobject(                                               │
│    163         self, submobject: Mobject, starting_submobject: Mobject, alpha: float │
│ File                                                                                 │
│"/home/emonhossain/.local/lib/python3.8/site-packages/manim/animation/creation.py",   │
│line 271, in interpolate_submobject                                                   │
│    268         index, subalpha = integer_interpolate(0, 2, alpha)                    │
│    269         if index == 0:                                                        │
│    270             submobject.pointwise_become_partial(outline, 0, subalpha)         │
│  ❱ 271             submobject.match_style(outline)                                   │
│    272         else:                                                                 │
│    273             submobject.interpolate(outline, starting_submobject, subalpha)    │
│    274                                                                               │
│ File "/home/emonhossain/.local/lib/python3.8/site-packages/manim/mobject/types/vector│
│ized_mobject.py", line 281, in match_style                                            │
│     278         return ret                                                           │
│     279                                                                              │
│     280     def match_style(self, vmobject, family=True):                            │
│  ❱  281         self.set_style(**vmobject.get_style(), family=False)                 │
│     282                                                                              │
│     283         if family:                                                           │
│     284             # Does its best to match up submobject lists, and                │
╰──────────────────────────────────────────────────────────────────────────────────────╯
TypeError: set_style() got an unexpected keyword argument 'stroke_opacity'
```

</details>

## System specifications

<details><summary>System Details</summary>

- OS (with version, e.g Windows 10 v2004 or macOS 10.15 (Catalina)): Ubuntu 20.04
- RAM: 8 GB
- Python version (`python/py/python3 --version`): Python 3.8.5
- Installed modules (provide output from `pip list`):
```
Package                Version
---------------------- --------------------
appdirs                1.4.3
apturl                 0.5.2
argon2-cffi            20.1.0
async-generator        1.10
attrs                  20.3.0
backcall               0.2.0
bcrypt                 3.1.7
beautifulsoup4         4.8.2
bleach                 3.3.0
blessings              1.7
blinker                1.4
bpython                0.21
Brlapi                 0.7.0
certifi                2019.11.28
cffi                   1.14.5
chardet                3.0.4
chrome-gnome-shell     0.0.0
Click                  7.0
cloudpickle            1.6.0
colorama               0.4.3
colour                 0.1.5
command-not-found      0.3
commonmark             0.9.1
cryptography           2.8
cupshelpers            1.0
curtsies               0.3.5
cwcwidth               0.1.4
cycler                 0.10.0
dbus-python            1.2.16
decorator              4.4.2
defer                  1.0.6
defusedxml             0.7.1
distlib                0.3.0
distro                 1.4.0
distro-info            0.23ubuntu1
duplicity              0.8.12.0
entrypoints            0.3
fasteners              0.14.1
filelock               3.0.12
future                 0.18.2
glcontext              2.3.2
gpg                    1.13.1-unknown
greenlet               1.0.0
html5lib               1.0.1
httplib2               0.14.0
idna                   2.8
imageio                2.9.0
importlib-metadata     1.5.0
ipykernel              5.5.3
ipython                7.20.0
ipython-genutils       0.2.0
ipywidgets             7.6.3
jedi                   0.18.0
Jinja2                 2.11.3
jsonschema             3.2.0
jupyter                1.0.0
jupyter-client         6.1.12
jupyter-console        6.4.0
jupyter-core           4.7.1
jupyterlab-pygments    0.1.2
jupyterlab-widgets     1.0.0
keyring                18.0.1
kiwisolver             1.3.1
language-selector      0.1
launchpadlib           1.10.13
lazr.restfulclient     0.14.2
lazr.uri               1.0.3
linecache2             1.0.0
lockfile               0.12.2
louis                  3.12.0
lxml                   4.5.0
macaroonbakery         1.3.1
Mako                   1.1.0
manim                  0.5.0
manim-fonts            0.1.0.post4
manim-rubikscube       0.0.8
ManimPango             0.2.4
mapbox-earcut          0.12.10
MarkupSafe             1.1.0
mathinspector          0.9.3.post0
matplotlib             3.3.4
mistune                0.8.4
moderngl               5.6.4
moderngl-window        2.3.0
monotonic              1.5
more-itertools         4.2.0
mpmath                 1.2.1
multipledispatch       0.6.0
nbclient               0.5.3
nbconvert              6.0.7
nbformat               5.1.3
nest-asyncio           1.5.1
netifaces              0.10.4
networkx               2.5
notebook               6.3.0
numpy                  1.20.1
oauthlib               3.1.0
olefile                0.46
opencv-python          4.5.1.48
packaging              20.9
pandocfilters          1.4.3
paramiko               2.6.0
parso                  0.8.1
pbr                    5.4.5
pexpect                4.6.0
pickleshare            0.7.5
Pillow                 7.0.0
pip                    21.0.1
progressbar            2.5
prometheus-client      0.10.1
prompt-toolkit         3.0.16
protobuf               3.6.1
ptyprocess             0.7.0
pycairo                1.20.0
pycparser              2.20
pycups                 1.9.73
pydub                  0.24.1
pygame                 2.0.1
pyglet                 1.5.15
PyGLM                  1.99.3
Pygments               2.8.0
PyGObject              3.36.0
PyJWT                  1.7.1
pymacaroons            0.13.0
PyNaCl                 1.3.0
PyOpenGL               3.1.5
PyOpenGL-accelerate    3.1.5
pyparsing              2.4.7
PyQt5                  5.14.1
pyRFC3339              1.1
pyrr                   0.10.3
pyrsistent             0.17.3
PySocks                1.6.8
python-apt             2.0.0+ubuntu0.20.4.4
python-dateutil        2.7.3
python-debian          0.1.36ubuntu1
pytz                   2019.3
PyWavelets             1.1.1
pyxdg                  0.26
PyYAML                 5.3.1
pyzmq                  22.0.3
qtconsole              5.0.3
QtPy                   1.9.0
reportlab              3.5.34
requests               2.22.0
requests-unixsocket    0.2.0
rich                   6.2.0
scikit-image           0.18.1
scipy                  1.6.1
scour                  0.37
screeninfo             0.6.7
SecretStorage          2.3.1
Send2Trash             1.5.0
setuptools             53.0.0
simplejson             3.16.0
sip                    4.19.21
six                    1.14.0
soupsieve              1.9.5
sympy                  1.7.1
systemd-python         234
terminado              0.9.4
testpath               0.4.4
testresources          2.0.0
tifffile               2021.3.17
torbrowser-launcher    0.3.2
tornado                6.1
tqdm                   4.57.0
traceback2             1.4.0
traitlets              5.0.5
ttkthemes              2.0.6
typing-extensions      3.7.4.3
ubuntu-advantage-tools 20.3
ubuntu-cleaner         1.1.3
ubuntu-drivers-common  0.0.0
ufw                    0.36
unattended-upgrades    0.1
unittest2              1.1.0
urllib3                1.25.8
usb-creator            0.3.7
validators             0.18.2
virtualenv             20.0.17
wadllib                1.3.3
watchdog               2.0.2
wcwidth                0.2.5
webencodings           0.5.1
wheel                  0.36.2
widgetsnbextension     3.5.1
xkit                   0.0.0
zipp                   1.0.0
```
</details>

<details><summary>LaTeX details</summary>

+ LaTeX distribution (e.g. TeX Live 2020):
+ Installed LaTeX packages:

</details>

<details><summary>FFMPEG</summary>

Output of `ffmpeg -version`:

```
ffmpeg version 4.2.4-1ubuntu0.1 Copyright (c) 2000-2020 the FFmpeg developers
built with gcc 9 (Ubuntu 9.3.0-10ubuntu2)
configuration: --prefix=/usr --extra-version=1ubuntu0.1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-nvenc --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
libavutil      56. 31.100 / 56. 31.100
libavcodec     58. 54.100 / 58. 54.100
libavformat    58. 29.100 / 58. 29.100
libavdevice    58.  8.100 / 58.  8.100
libavfilter     7. 57.100 /  7. 57.100
libavresample   4.  0.  0 /  4.  0.  0
libswscale      5.  5.100 /  5.  5.100
libswresample   3.  5.100 /  3.  5.100
libpostproc    55.  5.100 / 55.  5.100
```
</details>

**Comment by behackl** [comment]
Thank you for the report! The issue definitely lies within `VectorScene`, and in particular the fact that `self.label_vector` adds a `BackgroundRectangle` behind the vector, which in turn has a fairly particular `set_style` method which doesn't handle all passed keyword arguments.

I don't see an easy workaround without modifying the library code; one could try to add `**kwargs` to both the method head and the call to `VMobject.set_style`, but that might also break other code in the library.

**Comment by emonhossainraihan** [comment]
I see. Is there any manim version or pull request which didn't contain this behavior? @behackl. I am not super familiar with python. That's why maybe it will be hard for me to tweak with the library source code.  And most important, I am doing so many things with `VectorScene` that's why I need a solution.

**Comment by emonhossainraihan** [comment]
Does upcoming version v0.6.0 will also have that issue? @behackl

**Comment by behackl** [comment]
Fixed via #2658
